alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultProviderDataAccessor.java

```diff
@@ -177,6 +177,7 @@ private void mapUsersToProjectByEmail(Long providerConfigId, String projectHref,
         logger.debug("Removing {} user relations from project {} ", userRelationsToRemove.size(), project.getName());
         providerUserProjectRelationRepository.saveAll(userRelationsToAdd);
         providerUserProjectRelationRepository.deleteAll(userRelationsToRemove);
+        providerUserProjectRelationRepository.flush();
```

Comment:
What happens to the transaction when flush is called?

Reply:
The transaction still protects us. 
https://stackoverflow.com/questions/9908082/does-a-hibernate-transaction-rollback-delete-session-flushed-entities